rmf_fleet_adapter/src/rmf_fleet_adapter/TaskManager.hpp

```diff
@@ -190,13 +231,19 @@ class TaskManager : public std::enable_shared_from_this<TaskManager>
   friend class ActiveTask;
 
   agv::RobotContextPtr _context;
-  std::weak_ptr<BroadcastClient> _broadcast_client;
+  std::optional<std::weak_ptr<BroadcastClient>> _broadcast_client;
```

Comment:
I'm curious, what's the motive behind having `std::optional<std::weak_ptr<T>>` here? An ordinary `std::weak_ptr` is nullable, so what extra layer of expression do we achieve by wrapping a `std::optional` around it? I don't think there's anything wrong with it; I'm just not sure what it's meant to express.

Reply:
I was trying to come up with a way to distinguish the situation where we we get a `nullptr` when we try to lock the `weak_ptr` vs we're able to lock, but the `BroadcastClient` was never initialized (server_uri not provided) and hence is a `nullptr`. If there is a more elegant way to do this, please let me know.

Reply:
Ah that makes perfect sense, thanks for the explanation :+1:

Reply:
Oh, some more thoughts, though... Why would it be a `std::weak_ptr` to the broadcast client? Generally `std::weak_ptr` is used to avoid circular dependencies or in cases where something just wants to observe the lifetime of a resource without interfering with its lifecycle. I don't think either of those is the case here. So maybe this should just generally be a `std::shared_ptr<BroadcastClient>`?

If we have cases where there might be a delay in the availability of a `BroadcastClient` instance, then we can consider using `std::shared_ptr<std::shared_ptr<BroadcastClient>>` so that the inner `std::shared_ptr` can be updated at any time, and everything holding the outer `std::shared_ptr` will automatically get access to the new resource.

These aren't things we need to worry about for this release, but maybe thoughts to consider for the next iteration.

Reply:
Whoops sorry I merged it too quickly.

Ah yeah the rationale was to deal with the delay in availability of the `BraodcastClient`. I will add a comment based on what you mentioned as a TODO for future reference 👍🏼